orttraining/orttraining/python/training/ortmodule/_graph_execution_manager.py

```diff
@@ -61,6 +61,12 @@ def __init__(self, module):
         # TrainingAgent or InferenceAgent
         self._execution_agent = None
 
+        # enable fast path to skip some repeated logic if user knows that there will be no change for the entire training
```

Comment:
May we list here the implications of ```fast_path```. For example, when ```fast_path``` is enabled, x, y, z checks are not done? This will come in handy when we want to document the ```fast_path``` and when the user should use it.

Reply:
"some repeated logic" is too vague and can hide bugs. Be explicit on what is skipped under which circumstances